Grbl_Esp32/src/GCode.cpp

```diff
@@ -453,12 +453,18 @@ uint8_t gc_execute_line(char* line, uint8_t client) {
                     case 9:
                         switch (int_value) {
 #ifdef COOLANT_MIST_PIN
-                            case 7: gc_block.modal.coolant = CoolantMode::Mist; break;
+                            case 7:
+                                gc_block.modal.coolant      = {};
```

Comment:
Strictly speaking I think the = {} is unnecessary because gc_block is memset to 0 at the top - but it doesn't hurt apart from an insignificant performance hit.
The overall logic is crazy complicated. M7, M8, and M9 are in the same modal group so it is an error to have more than one per line - checked later.  After that check, if gc_block.modal.coolant is "disabled", gc_state.modal.coolant is also set to disabled, otherwise the single bit that is set in gc_block.modal.coolant is OR-ed into gc_state.modal.coolant.

Reply:
Probably. The compiler will probably get rid of it if its truely redundant.

Yes its too complicated for my taste as well. I would have expected just the flag to be set. Simple, uncomplicated, no or,... Apparently not... Thats for another day.